filebeat/docs/autodiscover-hints.asciidoc

```diff
@@ -278,31 +281,81 @@ filebeat.autodiscover:
     - type: nomad
       hints.enabled: true
       hints.default_config:
-        type: nomad
+        type: log
         paths:
-          - /var/lib/nomad/alloc/${data.nomad.allocation.id}/alloc/logs/${data.nomad.task.name}.*
+          - /opt/nomad/alloc/${data.nomad.allocation.id}/alloc/logs/${data.nomad.task.name}.*
 -----
 
-You can also disable default settings entirely, so only Jobs annotated like `co.elastic.logs/enabled: true`
-will be retrieved:
+You can also disable the default config such that only logs from jobs explicitly
+annotated with `"co.elastic.logs/enabled" = "true"` will be collected:
 
 [source,yaml]
 -----
 filebeat.autodiscover:
   providers:
     - type: nomad
       hints.enabled: true
-      hints.default_config.enabled: false
+      hints.default_config:
+        enabled: false
+        type: log
+        paths:
+          - /opt/nomad/alloc/${data.nomad.allocation.id}/alloc/logs/${data.nomad.task.name}.*
```

Comment:
I have a semi-related note about this path template (sorry if slightly off-topic)- when testing out this feature I noticed that the `task.name` data was only being published on the `meta.nomad` field of the event, not the `nomad` field used in this example. Here's the relevant code: https://github.com/elastic/beats/blob/076e0a693c18617454388c6759c4010e821c0f59/x-pack/libbeat/autodiscover/providers/nomad/nomad.go#L197-L210
In my tests, `${data.nomad.task.name}` gave an invalid template reference error, but `${data.meta.nomad.task.name}` resolved properly. Does this sound correct and if so, should all the documentation examples be updated to reflect this working usage? Or does a separate PR to fix the existing examples make more sense?

Reply:
I noted that there was a lot of duplicate information in the autodiscover events that get published by that code. Here's a sample document.

Reply:
> Can you open a separate issue please?

Opened #26538.

> But I wasn't sure what the expected behavior was.

It turns out there was some discussion around the need for this duplicate information in the original PR- see https://github.com/elastic/beats/pull/14954#discussion_r398699442.

Reply:
Great find. So based on that comment it sounds like we need to keep both `meta` and `nomad`. But we can make sure all the documented keys are present under the `nomad.`

Reply:
Good catch!

I think only the `nomad` fields need to be available for config templates. The meta fields are used for enrichment, but shouldn't be necessarily available for templates. We probably messed something up in the original PR, I remember going back and forth on this :grimacing: 

We also have to double-check that hints based configs have the same fields available.